### PR TITLE
[Routing Table] result table research

### PR DESCRIPTION
**Pull request by BiliouriV**
# Comments for current PR
* *add your comments here*
*  ...
 

<br><br />
<br><br />
               *!!! WARNING : DO NOT EDIT THE CHECKLIST BELOW !!!*
# Checklist : Merge to DEVELOP branch
## Code Quality
- [ ] Each function has a header (description, inputs, outputs) 
- [ ] Code is commented (particularly in hard to understand areas)
- [ ] There are no new warnings that can be corrected
- [ ] Commits policy is respected (constitancy commits, clear commits comments)
<br><br />
## Tests Review
- [ ]  Manual Tests for new features have been *reviewed*
- [ ]  Automatic Tests for new features have been *reviewed*
<br><br />
## Tests are passed the best as I can
- [ ]  Non regression existing tests are *passed OK*
- [ ]  New tests for new features & bug fixes are *passed OK*
<br><br />
## Documentation
- [ ] Documentation is up to date
- [ ] Changelog for next release is up-to-date : [fill here](https://github.com/Luos-io/Luos/pull/115)
- [ ] If needed, tests are briefly described in [test repository](https://github.com/Luos-io/tests)

### CODE REVIEW

File: inc/routing_table.h [hunk 1]
```diff
@@ -93,4 +103,9 @@ uint16_t RoutingTB_GetLastService(void);
 uint16_t *RoutingTB_GetLastNode(void);
 uint16_t RoutingTB_GetLastEntry(void);
 
+// ********************* research routing table ********************************
+search_result_t *RTFilter_Type(search_result_t *result, luos_type_t type);
```
Comment (nicolas-rabault) [inline inc/routing_table.h:107]: If you put a `search_result_t *` as an argument, why do you return one too?
as I see it user should be able to do:

``` C
search_result_t result;
RTFilter_Type(&result, STATE);
```
And get result on the result var.

File: inc/routing_table.h [hunk 2]
```diff
@@ -93,4 +103,9 @@ uint16_t RoutingTB_GetLastService(void);
...
+search_result_t *RTFilter_Node(search_result_t *result, uint16_t node_id);
+uint16_t RTFilter_ExtarctServiceID(search_result_t *result, uint16_t *id_table);
```
Comment (nicolas-rabault) [inline inc/routing_table.h:109]: Following my previous feedback,
Why users just doesn't do:
```C
result. result_table[0].id;
```

**Review by nicolas-rabault — COMMENTED** [review]
I just checked your .h

File: inc/routing_table.h (hunk 2 above)
Comment (BiliouriV) [reply inc/routing_table.h:109]: they can, but that is to say users need to know how searching_result_t as well as routing_table_t work.

Comment (nicolas-rabault) [reply inc/routing_table.h:109]: They will copy-paste it.
If the line is clear they will get it easily.
To make it smoother you could replace the result_table as table or list to be able to make
```C
result. list[0].id;
```

Comment (BiliouriV) [reply inc/routing_table.h:107]: but they will need to check the mode of this entry in the list. if it is a service or a node and which id they want. For example when we do RTFilter_Type() we keep all the services of the needed type and all the entries for their nodes in order to be able to understand in which node they are... The user should apply a loop and search by mode = SERVICE to find their ids...

File: inc/routing_table.h (hunk 1 above)
Comment (BiliouriV) [reply inc/routing_table.h:105]: we can do it like this, but we will need to call the functions serially and not like this: 
`RTFilter_Type(RTFilter_Node(0, 1), TYPE)`

Comment (nicolas-rabault) [reply inc/routing_table.h:105]: Yes. 
Perhaps we can do both by keeping your function like that but user need to pass the search_result_t (0 is not allowed)

File: inc/routing_table.h (hunk 2 above)
Comment (nicolas-rabault) [reply inc/routing_table.h:107]: I don't think your research function will have to return node entry. only services.

**Pull request closed without merging**